calypso/service.go

```diff
@@ -82,47 +103,241 @@ type vData struct {
 	Signature *darc.Signature
 }
 
+// ProcessClientRequest implements onet.Service. We override the version
+// we normally get from embeddeding onet.ServiceProcessor in order to
+// hook it and get a look at the http.Request.
+func (s *Service) ProcessClientRequest(req *http.Request, path string, buf []byte) ([]byte, *onet.StreamingTunnel, error) {
+	if path == "Authorise" {
```

Comment:
you could just move the `disableLoopbackCheck` here: e.g., `!disableLoopbackCheck && path == "Authorise"`